spec/models/uploaded_file_spec.rb

```diff
@@ -0,0 +1,110 @@
+# Much of this testing is the complex configurable stuff underneath UploadedFile
+# and simulating the AWS configuration that would be in production.
+
+# rubocop:disable Metrics/LineLength
+RSpec.describe 'Hyrax::UploadedFile' do # rubocop:disable RSpec/DescribeClass
+  let(:file) { File.open(fixture_path + '/images/world.png') }
+  let(:upload) { Hyrax::UploadedFile.create(file: file) }
+  let(:config) { CarrierWave.configure { |c| c } }
+  let(:bucket_name) { 'hyku-carrierwave-test' }
+  let(:bigurl) { 'https://demo-application-w5twjcsuyx3v-uploadbucket-4l1zpvmp62nf.s3.amazonaws.com//var/app/current/tmp/uploads/hyrax/uploaded_file/file/31/image.png?X-Amz-Expires=600\u0026X-Amz-Date=20161214T193306Z\u0026X-Amz-Security-Token=FQoDYXdzEMT//////////wEaDMj3NLTbn4Y3JgbItSK3A57LyrcXBHQlP6lM7cT/2N9naRgRSef4EG/AxCCjMGcEVdt4X5ZsfHdzNiD6L0GODXmrP3quoXNBNZCoUVo3DY5E0P67iz9tYC2Ac%2BILJ%2BBzELNz84XI7C9zg6CCecZ8oeNjCTJXsMZ3xLx2bN099sl%2BY5nduDXAxen2Z63QKw7kiuuEXin/z%2B4ywFSP/Z1Sqbjkq4Qwjs5FUSyyz61wjl1%2Bg8uIJ5u3HTOlb8eZpk7gUCtdmLIE7mK1eZe5azUJC8XBW7Eu7jaRyM2PKMwjVnwepnfgPyEDqJSzKYJt1bGXgnQEN7logEKNOjmOcJqggM5Tc7PD40USAveIQ6E8ny/X0N%2BZ/X1rZTaCiAH1aWwVNqa0M43mlECrBeDv9I9BRMJzp4btvEgHKODrJe2MawDu4L1%2BzVNgOD7TZjrFt9zSEpyQK79dh8oHuyzDL0C%2Bpw3zL2ambsJ5OX6UnMuAmrkBbin1PKh2nHFkL/0xXAb2ZbSV6vKBxzKeQ62HMvv8UqypKbkwOMnstxyGGp00r6m6vL62x%2BTDergiiRfs947NyfJnP5l/rNRNMNesGo6kBmAqpACaBPAo0Z3GwgU%3D\u0026X-Amz-Algorithm=AWS4-HMAC-SHA256\u0026X-Amz-Credential=ASIAIB72YBSAAINUZRPQ/20161214/us-east-1/s3/aws4_request\u0026X-Amz-SignedHeaders=host\u0026X-Amz-Signature=f9bfb2b8d6114bccb6e88e4c0526bf19c5658b587ee368d04b42b1881d5359db' }
+
+  describe 'documented problem with carrierwave/fog' do
+    describe CarrierWave::SanitizedFile do
+      it 'cannot handle S3 URI' do
+        # if the following trips, it means we *might* be able to use just carrierwave/fog again, w/o carrierwave-aws
```

Comment:
Seems obvious: because the test would be passing in the same way as the tests for the other classes.  There is more than one bug in the library though, so just because this one codepath became viable would not mean that switching back would be gold.  Also, carrierwave-aws seems like it will be superior anyway.